src/irmin-pack/inode.ml

```diff
@@ -106,6 +106,15 @@ struct
       |~ case1 "Tree" tree_t (fun t -> Tree t)
       |> sealv
 
+    let v_t =
+      if not Conf.prefix_unsable_pre_hash then v_t
+      else
+        let pre_hash : v Irmin.Type.encode_bin =
+          Irmin.Type.(
+            pre_hash (map (pair char v_t) (fun (_, s) -> s) (fun s -> ('i', s))))
+        in
+        Irmin.Type.like ~pre_hash v_t
```

Comment:
I'm not sure we need to do this for inodes. They are already prefixed by their variant number (we just need to make sure blobs and commits do not have the same prefix).

Ideally we should also have a mechanism to statically check that all the prefixes are different. Having a startup check would already be great.